src/core/namespace.c

```diff
@@ -1358,6 +1403,15 @@ static int mount_image(
                 return log_debug_errno(r, "Failed to extract extension name from %s: %m", mount_entry_source(m));
 
         if (m->mode == MOUNT_EXTENSION_IMAGE) {
+                assert(extension_verities);
+                assert(n_extension_verities);
+
+                /* If this is a dm-verity image, check for duplicates, as Overlayfs rejects them as
+                 * explained above. */
+                r = verities_find(mount_entry_source(m), extension_verities, n_extension_verities, &verity);
+                if (r <= 0)
+                        return r;
```

Comment:
this is is semi ugly I'd say, because you insert this data into the array already before you know the function will actually really mount an image. I mean, there are exit paths to this function where we don#t fail fatally but don't actually mount anything. It would be cleaner to only insert the thing to the array when we actually keep the thing around